environs/config/config_test.go

```diff
@@ -669,104 +689,104 @@ func (test configTest) check(c *gc.C) {
+		c.Check(ok, jc.IsFalse)
+		c.Check(agentVersion, gc.Equals, version.Zero)
 	}
 
 	if expected, ok := test.attrs["uuid"]; ok {
-		c.Assert(cfg.UUID(), gc.Equals, expected)
+		c.Check(cfg.UUID(), gc.Equals, expected)
 	}
 
 	dev, _ := test.attrs["development"].(bool)
-	c.Assert(cfg.Development(), gc.Equals, dev)
+	c.Check(cfg.Development(), gc.Equals, dev)
 
 	seriesAttr, _ := test.attrs["default-series"].(string)
 	defaultSeries, ok := cfg.DefaultSeries()
 	if seriesAttr != "" {
-		c.Assert(ok, jc.IsTrue)
-		c.Assert(defaultSeries, gc.Equals, seriesAttr)
+		c.Check(ok, jc.IsTrue)
+		c.Check(defaultSeries, gc.Equals, seriesAttr)
 	} else {
-		c.Assert(ok, jc.IsFalse)
-		c.Assert(defaultSeries, gc.Equals, "")
+		c.Check(ok, jc.IsFalse)
+		c.Check(defaultSeries, gc.Equals, "")
 	}
 
 	if m, _ := test.attrs["firewall-mode"].(string); m != "" {
-		c.Assert(cfg.FirewallMode(), gc.Equals, m)
+		c.Check(cfg.FirewallMode(), gc.Equals, m)
 	}
 
 	if m, _ := test.attrs["default-space"].(string); m != "" {
-		c.Assert(cfg.DefaultSpace(), gc.Equals, m)
+		c.Check(cfg.DefaultSpace(), gc.Equals, m)
 	}
 
 	keys, _ := test.attrs["authorized-keys"].(string)
-	c.Assert(cfg.AuthorizedKeys(), gc.Equals, keys)
+	c.Check(cfg.AuthorizedKeys(), gc.Equals, keys)
 
 	lfCfg, hasLogCfg := cfg.LogFwdSyslog()
 	if v, ok := test.attrs["logforward-enabled"].(bool); ok {
-		c.Assert(hasLogCfg, jc.IsTrue)
-		c.Assert(lfCfg.Enabled, gc.Equals, v)
+		c.Check(hasLogCfg, jc.IsTrue)
+		c.Check(lfCfg.Enabled, gc.Equals, v)
 	}
 	if v, ok := test.attrs["syslog-ca-cert"].(string); v != "" {
-		c.Assert(hasLogCfg, jc.IsTrue)
-		c.Assert(lfCfg.CACert, gc.Equals, v)
+		c.Check(hasLogCfg, jc.IsTrue)
+		c.Check(lfCfg.CACert, gc.Equals, v)
 	} else if ok {
-		c.Assert(hasLogCfg, jc.IsTrue)
+		c.Check(hasLogCfg, jc.IsTrue)
 		c.Check(lfCfg.CACert, gc.Equals, "")
 	}
 	if v, ok := test.attrs["syslog-client-cert"].(string); v != "" {
-		c.Assert(hasLogCfg, jc.IsTrue)
-		c.Assert(lfCfg.ClientCert, gc.Equals, v)
+		c.Check(hasLogCfg, jc.IsTrue)
+		c.Check(lfCfg.ClientCert, gc.Equals, v)
 	} else if ok {
-		c.Assert(hasLogCfg, jc.IsTrue)
+		c.Check(hasLogCfg, jc.IsTrue)
 		c.Check(lfCfg.ClientCert, gc.Equals, "")
 	}
 	if v, ok := test.attrs["syslog-client-key"].(string); v != "" {
-		c.Assert(hasLogCfg, jc.IsTrue)
-		c.Assert(lfCfg.ClientKey, gc.Equals, v)
+		c.Check(hasLogCfg, jc.IsTrue)
+		c.Check(lfCfg.ClientKey, gc.Equals, v)
 	} else if ok {
-		c.Assert(hasLogCfg, jc.IsTrue)
+		c.Check(hasLogCfg, jc.IsTrue)
 		c.Check(lfCfg.ClientKey, gc.Equals, "")
 	}
 
 	if v, ok := test.attrs["ssl-hostname-verification"]; ok {
-		c.Assert(cfg.SSLHostnameVerification(), gc.Equals, v)
+		c.Check(cfg.SSLHostnameVerification(), gc.Equals, v)
 	}
 
 	if v, ok := test.attrs["provisioner-harvest-mode"]; ok {
 		harvestMeth, err := config.ParseHarvestMode(v.(string))
-		c.Assert(err, jc.ErrorIsNil)
-		c.Assert(cfg.ProvisionerHarvestMode(), gc.Equals, harvestMeth)
+		c.Check(err, jc.ErrorIsNil)
```

Comment:
I'm not sure if you should use Check for ErrorIsNil, as often when error isn't nil it means the return object may be wrong, which can cause nil panics.

Reply:
I'll double check - in the first case of running check on an err, if the err wasn't nil, the method was made to return so we wouldn't use it but continue the tests. I'll double check the rest.

Reply:
In this case, if ParseHarvestMode returns an error, there is no panic as the other return value is a string. It's been verified